Userland/Services/ChessEngine/main.cpp

```diff
@@ -14,7 +14,7 @@ ErrorOr<int> serenity_main(Main::Arguments)
 {
     TRY(Core::System::pledge("stdio recvfd sendfd unix rpath"));
     Core::EventLoop loop;
-    TRY(Core::System::pledge("stdio recvfd sendfd unix"));
+    TRY(Core::System::revoke("rpath"));
```

Comment:
Maybe I'm biased after having stared at a bunch of pledge code, but having to switch my thinking to "what this process cant do anymore" instead of just having a list of things it CAN do now... is a bit unfortunate.

IMHO, it seems like having the list of things a process can still do is far more useful from a readability / security posture perspective. Maybe I'm missing something though, interested to hear what others have to say.

Reply:
This is a good point. A single `pledge()` call lets you see the remaining promises easily. But with `revoke()` you can actually easily see permissions being dropped after being used. This helped me find #11207. 

Workaround 1: I think you can see active promises in the System Monitor? We could also make the `pledge()` syscall debug-print out active promises after being called.

Workaround 2: combine both functions into one:
```cpp
pledge("stdio rpath unix inet");

// ... do stuff ...

revoke_but_keep( Revoke("unix inet"), Keep("stdio rpath") );
```
(This seems like chaos)

Reply:
I also had the idea of a `revokeExpect()` function (https://github.com/SerenityOS/serenity/issues/11140#issuecomment-986687522) that would allow you to enforce the number of leftover promises.

In all, enforcing an application to have an initial set of promises (https://github.com/SerenityOS/serenity/pull/11209#discussion_r767146812) and revoking them after use might make it easier to track what's going on.

In the current patch there are already times (I think) the `signal` promise was only being dropped together with `unix` out of "laziness"? If we make dropping permissions the 'fun path', it encourages programmers to do so often.

Reply:
As for readability/security I now have a strong opinion: https://github.com/SerenityOS/serenity/pull/11209#discussion_r767164099